umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/platform.clj

```diff
@@ -15,29 +16,52 @@
-        instruments (->> (concat
-                          (map (partial get instruments-mapping) instrument-ids)
-                          (filter #(= platform-id (:mounted-on-id %)) (vals instruments-mapping)))
-                         (map #(dissoc % :mounted-on-id))
-                         distinct
-                         seq)]
-    (util/remove-nil-keys
-      {:ShortName (value-of platform-elem iso/short-name-xpath)
-       :LongName (value-of platform-elem iso/long-name-xpath)
-       :Type (without-default-value-of platform-elem "gmi:description/gco:CharacterString")
-       :Characteristics (char-and-opsmode/parse-characteristics platform-elem)
-       :Instruments instruments})))
+  ([doc base-xpath instruments-mapping platform-elem]
+   (let [platform-id (get-in platform-elem [:attrs :id])
+         instrument-ids (keep #(get-in % [:attrs :xlink/href]) (select platform-elem "gmi:instrument"))
+         instruments (->> (concat
+                            (map (partial get instruments-mapping) instrument-ids)
+                            (filter #(= platform-id (:mounted-on-id %)) (vals instruments-mapping)))
+                           (map #(dissoc % :mounted-on-id))
+                           distinct
+                           seq)]
+     (util/remove-nil-keys
+       {:ShortName (value-of platform-elem iso/short-name-xpath)
+        :LongName (value-of platform-elem iso/long-name-xpath)
+        :Type (without-default-value-of platform-elem "gmi:description/gco:CharacterString")
+        :Characteristics (char-and-opsmode/parse-characteristics platform-elem)
+        :Instruments instruments})))
+   ([doc base-xpath platform-elem]
+    ;; This is the case when platform-elem is from alternative path. This platform will only contain ShortName and LongName.
+    (when-let [short-long-name (value-of platform-elem iso/short-name-xpath)]
+     (let [short-long-name-list (string/split short-long-name #">")]
+       (util/remove-nil-keys
+         {:ShortName (string/trim (first short-long-name-list))
+          :LongName (when-let [long-name (second short-long-name-list)]
+                      (string/trim long-name))})))))
 
 (defn parse-platforms
   "Returns the platforms parsed from the given xml document."
-  [doc base-xpath sanitize?]
-  (let [instruments-mapping (inst/xml-elem->instruments-mapping doc base-xpath)
-        platforms (if (or (map? instruments-mapping) (nil? (seq instruments-mapping)))
-                    (seq (map #(xml-elem->platform doc base-xpath instruments-mapping %)
-                              (select doc (str base-xpath platforms-xpath))))
-                    ;; NOAA case when instruments are not associated with any platforms.
-                    (seq (map #(assoc % :Instruments instruments-mapping) su/not-provided-platforms)))]
-    (or (seq platforms) (when sanitize? su/not-provided-platforms))))
+  ([doc base-xpath sanitize?]
```

Comment:
If we are splitting this function isomends vs. isosmap, does it really belong in the iso-shared anymore?  iso-shared implies that the implementation between the two iso is in fact shared.

Reply:
Good question. I thought about that. I am leaving them in the the iso_shared because the current implementation of platforms/instruments for ISOMENDS from alternative xpath is just temporary - as you can see, there are no relationship between them and that is gonna change. When it does, I think there will still be a lot of code in common.